### PR TITLE
fix: recording start time

### PR DESCRIPTION
**Pull request by pauldambra**
two changes in here

1. network events contain requests which have their own timestamps, those timestamps can be before the event timestamp, for sorting and picking zero time we now consider the request timestamps instead of the event timestamps
2. we already deduplicate events but sometimes we see duplicates that only differ by `delay` - I'd like to fix that at source but since that doesn't affect existing data anyway - we now ignore delay when deduplicating

this stops us duplicating some displays and stops us saying some network requests are 2 minutes before the recording

|before|after|
|--|--|
|Screenshot 2024-03-07 at 16 30 43|Screenshot 2024-03-07 at 16 30 13

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts [hunk 1]
```diff
@@ -742,9 +769,18 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
...
-                return meta?.start_time ? dayjs(meta.start_time) : undefined
+            (s) => [s.sessionPlayerMetaData, s.sessionPlayerSnapshotData],
+            (meta, sessionPlayerSnapshotData): Dayjs | undefined => {
+                // NOTE: We might end up with more snapshots than we knew about when we started the recording so we
+                // either use the metadata start point or the first snapshot, whichever is earlier.
+                const start = meta?.start_time ? dayjs(meta.start_time) : undefined
+                const snapshots = sessionPlayerSnapshotData?.snapshots || []
+                const firstEventTimestamp = getSnapshotSortingTimestamp(snapshots[0])
+                // eslint-disable-next-line no-console
+                console.log('[wat] picking start', start?.valueOf(), firstEventTimestamp, snapshots)
```
Comment (daibhin) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:780]: Do we need this log? Could be noisy

File: frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts [hunk 2]
```diff
@@ -134,6 +134,22 @@ function snapshotDescription(snapshot: eventWithTime): string {
...
+    start: Dayjs | undefined
+): {
+    timeInRecording: number
+    timestamp: dayjs.Dayjs
+} {
+    const timestamp = dayjs(thingWithTime.timestamp)
+    const timeInRecording = timestamp.valueOf() - (start?.valueOf() ?? 0)
+    if (timeInRecording < 0) {
+        // eslint-disable-next-line no-console
+        console.log('[wat] Negative time in recording', { timeInRecording, timestamp, start, thingWithTime })
```
Comment (daibhin) [inline frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts:148]: Should this be a `console.error`?

**Review by daibhin — APPROVED** [review]
wat

File: frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts (hunk 2 above)
Comment (marandaneto) [reply frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts:148]: Only if we generate those timestamps would be an error, if we read from somewhere, nothing we can do anyway.

Comment (pauldambra) [reply frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts:148]: ah, no, leftover from the draft PR 🙈

File: frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts [hunk 3]
```diff
@@ -134,6 +134,22 @@ function snapshotDescription(snapshot: eventWithTime): string {
...
+    }
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts:149]:
```suggestion
```

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts (hunk 1 above)
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:780]:
```suggestion
```

**Pull request merged**